### ISSUE TITLE
Add saving/loading to Neural Network module 

### ISSUE THREAD

**Issue by josephjaspers** [josephjaspers/BlackCat_Tensors#25]
need the trained net support save and load net,
as the net at https://github.com/Artelnics/OpenNN.main supprot function:1、

save the trained model
load the saved model trained to predict

**Comment by josephjaspers** [comment]
Added save/load
https://github.com/josephjaspers/BlackCat_Tensors/commit/520b2b71eff5e5772c8754a1282e97ff3affd9bc

(base) joseph@joseph-G751JY:~/BlackCat_Tensors/BlackCat_Tensors/examples/mnist_test/forward_net$ cat architecture.yaml 
FeedForward:
	inputs: 784
	outputs: 256
Logistic:
	inputs: 256
	outputs: 256
FeedForward:
	inputs: 256
	outputs: 10
SoftMax:
	inputs: 10
	outputs: 10
OutputLayer:
	inputs: 10
	outputs: 10
(base) joseph@joseph-G751JY:~/BlackCat_Tensors/BlackCat_Tensors/examples/mnist_test/forward_net$ ls l0_FeedForward/
b.vec  w.mat
(base) joseph@joseph-G751JY:~/BlackCat_Tensors/BlackCat_Tensors/examples/mnist_test/forward_net$ ls l2_FeedForward/
b.vec  w.mat

The '.vec' and '.mat' files are really just csv  files.   (The .vec files have only 1 row).

You may save a model by calling  "network.save(directory)"
You may load a model by calling "network.load(directory)" 

save will create a directory if it does not exist. 
This still needs to be tested on Windows.

**Comment by josephjaspers** [comment]
Added and tested.
